server/rest/src/main/java/org/infinispan/rest/search/InfinispanSearchRequest.java

```diff
@@ -59,10 +61,12 @@ private QueryRequest getQueryRequest() throws IOException {
       if (request.method() == HttpMethod.GET) {
          String queryString = getParameterValue(QUERY_STRING);
```

Comment:
The whole body of this ```if```statement would look better if extracted as a separate getQueryFromString method in the spirit of ```getQueryFromJSON```